src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -117,7 +121,7 @@ internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToCo
                         }
                     }
 
-                    ArrayPool<FoundPropertiesAsync>.Shared.Return(state.Current.CtorArgumentState!.FoundPropertiesAsync!, clearArray: true);
+                    ArrayPool<FoundPropertyAsync>.Shared.Return(argumentState.FoundPropertiesAsync!, clearArray: true);
```

Comment:
I recommend nulling out `argumentState.FoundPropertiesAsync` after the `Return`. Otherwise a potential bug could write to another user's pooled array.

Reply:
Done, thanks!